enumerable.rb

```diff
@@ -1,72 +1,87 @@
-module Enumerable
-    def my_each(arr)
-result=[]
-for i in 0..length
-    result.push(yield(arr[i]))
-end
-result
-    end
- 
-    def my_each_with_index(arr)
-        result = []
-        for i in 0..length
-            result.push(yield(arr[i],i))
-        end
-        result
-    end
-
-    def my_select
-        result=[]
-        my_each do |x|
-            output.push(x) if yield x
-        end
-        result
-    end
-
-    def my_all
-        result = true
-        my_each do |x|
-            result = false unless yield x
-        end
-        result
-    end
-
-    def my_any?
-        result = false
-        my_each do |x|
-            result = false if yield x
-        end 
-        result
-    end
-
-    def my_none?
-        result = true
-        my_each do |x|
-            result = false if yield x
-        end
-        result
-    end
-
-    def my_count
-        count = 0
-        my_each do |x|
-            count +=1 if yield x
-        end
-        count
-    end
-
-    def my_map(arr)
-        result = []
-        my_each(arr) do |x|
-            result.push(yield(x))
-        end
-        result
-    end
-
-
-
-
-
-
-
-end
\ No newline at end of file
+module Enumerable
+  def my_each(arr)
+    result = []
+    for i in 0..length
+      result.push(yield(arr[i]))
+    end
+    result
+  end
+
+  def my_each_with_index(arr)
+    result = []
+    for i in 0..length
+      result.push(yield(arr[i], i))
+    end
+    result
+  end
+
+  def my_select
+    result = []
+    my_each do |x|
+      output.push(x) if yield x
+    end
+    result
+  end
+
+  def my_all
+    result = true
+    my_each do |x|
+      result = false unless yield x
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each do |x|
+      result = false if yield x
+    end
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each do |x|
+      result = false if yield x
+    end
+    result
+  end
+
+  def my_count
+    count = 0
+    my_each do |x|
+      count += 1 if yield x
+    end
+    count
+  end
+
+  def my_map(arr)
+    result = []
+    my_each(arr) do |x|
+      result.push(yield(x))
+    end
+    result
+  end
+
+  def my_map(&block)
+    new_result = []
+    self.my_each do |x|
+      new_result.push(block.call(i))
+    end
+    new_result
+  end
+
+  def my_inject(initial = 0)
+    i = 0
+    total = initial
+    while (i < self.length)
+      total = yield(total, self[i])
+      i = i + 1
+    end
+    total
+  end
```

Comment:
- `my_inject` should be able to accept a `Symbol` as an operator (e.g `:+`, `:*`):
   ```ruby
   irb(main):055:0> [1,2,3].inject(:+)
   => 6
   irb(main):056:0> [1,2,3].my_inject(:+)
   LocalJumpError (no block given (yield))
   ```

- `my_inject` should be able to accept two arguments, first as an initial value and the other as a `Symbol`:

   ```ruby
   irb(main):057:0> [1,2,3].inject(5,:+)
   => 11
   irb(main):058:0> [1,2,3].my_inject(5,:+)
   LocalJumpError (no block given (yield))
   ```

### Find more details about [`inject?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject)